playroom/components/loader.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+
+type Props = {
+    loader: string | (() => Promise<any>);
+    render: (data: any) => React.ReactElement;
+    renderLoading: () => React.ReactElement;
+    renderError: () => React.ReactElement;
+};
+
+export const Loader: React.FC<Props> = ({loader, render, renderLoading, renderError}) => {
+    const [content, setContent] = React.useState<React.ReactElement>(<></>);
```

Comment:
it isn't a good idea to store react elements (jsx) inside a react state. I'd just store the api result and/or error in react state, and call `render` `renderLoading` or `renderError` at render time